File: dev-docs/snippets/blob-buckets/get-bucket.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Fetch one bucket by its bucketId or bucketKey.
export async function getBucket(client: JsBaoClient, bucketIdOrKey: string) {
  // #region example
  const bucket = await client.blobBuckets.getBucket(bucketIdOrKey);
  // #endregion example
  return bucket;
}

File: dev-docs/snippets/blob-buckets/get-bucket.swift
import JsBaoClient

// Fetch one bucket by its bucketId or bucketKey.
// Swift returns a typed `BlobBucketInfo`.
func getBucket(client: JsBaoClient, bucketIdOrKey: String) async throws {
  // #region example
  let bucket = try await client.blobBuckets.getBucket(bucketIdOrKey: bucketIdOrKey)
  // #endregion example
  _ = bucket
}
